spec/migrations/20170927135007_migrate_dialog_field_associations_to_use_new_relationship_spec.rb

```diff
@@ -0,0 +1,88 @@
+
+      migrate
+
+      expect(dialog_field_association_stub.count).to eq(0)
+    end
+
+    it "does not create a circular reference" do
+      dialog_field_stub.create!(:name => "dialog_field4", :trigger_auto_refresh => true, :auto_refresh => true, :position => 0, :dialog_group_id => dialog_group_id, :type => "DialogFieldRadioButton")
+      dialog_field_stub.create!(:name => "dialog_field1", :auto_refresh => true, :trigger_auto_refresh => true, :position => 4, :dialog_group_id => dialog_group_id, :type => "DialogFieldTagControl")
+      expect(dialog_field_association_stub.count).to eq(0)
+
+      migrate
+
+      expect(dialog_field_association_stub.count).to eq(1)
+      expect(dialog_field_association_stub.first.trigger_id).to eq(dialog_field_stub.first.id)
+      expect(dialog_field_association_stub.first.respond_id).to eq(dialog_field_stub.second.id)
+    end
+
+    it "does create a reference when valid one is present" do
+      dialog_field_stub.create!(:name => "dialog_field4", :auto_refresh => true, :position => 2, :dialog_group_id => dialog_group_id)
+      dialog_field_stub.create!(:name => "dialog_field1", :trigger_auto_refresh => true, :position => 0, :dialog_group_id => dialog_group_id)
+      expect(dialog_field_association_stub.count).to eq(0)
+
+      migrate
+
+      expect(dialog_field_association_stub.count).to eq(1)
+      expect(dialog_field_association_stub.first.trigger_id).to eq(dialog_field_stub.second.id)
+      expect(dialog_field_association_stub.first.respond_id).to eq(dialog_field_stub.first.id)
+    end
+
+    it "does not create association when dialog tab is missing" do
+      dialog_stub.create!
+      dialog_tab_stub.delete_all
+      dialog_group_stub.create!(:dialog_tab_id => 5, :position => 3)
+      dialog_field_stub.create!(:name => "dialog_field4", :auto_refresh => true, :position => 2, :dialog_group_id => dialog_group_stub.first.id)
+      dialog_field_stub.create!(:name => "dialog_field1", :trigger_auto_refresh => true, :position => 0, :dialog_group_id => dialog_group_stub.first.id)
+      expect(dialog_field_association_stub.count).to eq(0)
+
+      migrate
+
+      expect(dialog_field_association_stub.count).to eq(0)
+    end
+
+    it "only creates dialog associations if the fields are on the same dialog" do
+      dialog_stub.create!
+      dialog_tab_stub.create!(:dialog_id => dialog_stub.first.id, :position => 8)
+      dialog_group_stub.create!(:dialog_tab_id => dialog_tab_stub.first.id, :position => 7)
+      dialog_field_stub.create!(:name => "dialog_field4", :trigger_auto_refresh => true, :position => 4, :dialog_group_id => dialog_group_stub.first.id, :type => "DialogFieldDateControl")
+
+      expect(dialog_field_association_stub.count).to eq(0)
+
+      migrate
+
+      expect(dialog_field_association_stub.count).to eq(0)
+    end
+  end
+
+  migration_context :down do
+    it "should delete dialog field associations" do
+      migrate
```

Comment:
This should create a dialog_field_association record before the migrate, otherwise you are testing nothing.